Bottom: c52ae360d93ebff503db26841cb6a14fc5ad1179
Top:    6d82ee7b5d0edeb6718514a96783f4bcccf09470
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-01-24 03:24:44 +0000

Add profiling to file load









---

diff --git a/src/file.c b/src/file.c
index fb1c923..fb75652 100644
--- a/src/file.c
+++ b/src/file.c
@@ -350,6 +350,10 @@ LoadPCB (char *Filename)
 {
   PCBTypePtr newPCB = CreateNewPCB (False);
   Boolean units_mm;
+  clock_t start, end;
+  double elapsed;
+
+  start = clock();
 
   /* new data isn't added to the undo list */
   if (!ParsePCB (newPCB, Filename))
@@ -380,6 +384,10 @@ LoadPCB (char *Filename)
 	  CreateDefaultFont ();
 	}
 
+  end = clock();
+  elapsed = ((double) (end - start)) / CLOCKS_PER_SEC;
+  printf ("Loading file %s took %f\n", Filename, elapsed);
+
       /* clear 'changed flag' */
       SetChangedFlag (False);
       PCB->Filename = MyStrdup (Filename, "LoadPCB()");
